subsys/logging/backends/log_backend_semihost.c

```diff
@@ -0,0 +1,71 @@
+/* Copyright (c) 2024 Contributors to the logging subsystem.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <stdio.h>
+#include <stddef.h>
+#include <zephyr/logging/log_backend.h>
+#include <zephyr/logging/log_backend_std.h>
+#include <zephyr/logging/log_core.h>
+#include <zephyr/logging/log_output.h>
+#include <zephyr/arch/common/semihost.h>
+
+static uint8_t buf[CONFIG_LOG_BACKEND_SEMIHOST_BUFFER_SIZE] = {0};
```

Comment:
No need to init a `static` buffer.

```suggestion
static uint8_t buf[CONFIG_LOG_BACKEND_SEMIHOST_BUFFER_SIZE];
```